### PR TITLE
Add more ways to detect runtime directory

### PR DESCRIPTION
**Pull request by wojciechkepka**
Closes: #212

### CODE REVIEW

File: helix-core/src/lib.rs [hunk 1]
```diff
@@ -48,15 +51,26 @@ pub fn find_root(root: Option<&str>) -> Option<std::path::PathBuf> {
 
 #[cfg(not(embed_runtime))]
 pub fn runtime_dir() -> std::path::PathBuf {
```
Comment (archseer) [inline helix-core/src/lib.rs:53]: Can you make this private now? We always want to use the cached variable

File: helix-core/src/lib.rs [hunk 2]
```diff
@@ -48,15 +51,26 @@ pub fn find_root(root: Option<&str>) -> Option<std::path::PathBuf> {
...
+        return dir.into();
+    }
+
+    const RT_DIR: &str = "runtime";
+    let conf_dir = config_dir().join(RT_DIR);
+    if conf_dir.exists() {
+        return conf_dir;
+    }
+
+    if let Ok(dir) = std::env::var("CARGO_MANIFEST_DIR") {
```
Comment (archseer) [inline helix-core/src/lib.rs:64]: Nice! This is only via 'cargo' right?

Comment (wojciechkepka) [reply helix-core/src/lib.rs:64]: Yes, it’s exported only during cargo build/run.

File: helix-core/src/lib.rs (hunk 1 above)
Comment (wojciechkepka) [reply helix-core/src/lib.rs]: Done.

**Comment by archseer** [comment]
Sorry, can you also update the install part of the readme? The text regarding needing to set the var in development is no longer relevant, and for regular installs we can point out to the user that it's possible to either copy or symlink the `runtime/` dir under the config dir.

File: helix-core/src/lib.rs (hunk 2 above)
Comment (pickfire) [reply helix-core/src/lib.rs:64]: Good idea. I didn't thought about this, if we run from cargo we don't need to use the runtime dir with this.

Comment (pickfire) [reply helix-core/src/lib.rs:64]: Good idea. I didn't thought about this, if we run from cargo we don't need to use the runtime dir with this.

**Comment by wojciechkepka** [comment]
Good idea, updated the README. I also changed how the directory is picked up when using executable dir.  From README I understood that you can have this structure:
```
bin
└── hx
└── runtime
```

And in reality it expected the runtime directory to be "extracted" alongside of hx. So I changed it to use `runtime` directory instead. Should I revert it to use the parent directory?

File: README.md [hunk 3]
```diff
@@ -41,14 +41,12 @@ cargo install --path helix-term
...
-runtime inside the same folder as the executable, but that can be overriden via
-the `HELIX_RUNTIME` environment variable.
-
-> NOTE: You should set this to <path to repository>/runtime in development (if
-> running via cargo).
-> 
-> `export HELIX_RUNTIME=$PWD/runtime`
+Now copy the `runtime/` directory somewhere. Helix will by default look for the runtime
+inside the config directory or the same directory as executable, but that can be overriden
+via the `HELIX_RUNTIME` environment variable.
```
Comment (pickfire) [inline README.md:46]: Maybe you might want to mention that one can just soft-link the runtime to the git repo runtime to be easier?

**Comment by pickfire** [comment]
Looks good to me. We could suggest the soft link thing but not necessary for this patch.

**Review by pickfire — APPROVED** [review]

**Comment by archseer** [comment]
> And in reality it expected the runtime directory to be "extracted" alongside of hx. So I changed it to use `runtime` directory instead. Should I revert it to use the parent directory?

Good catch! ...this was certainly a bug and explains why there was issues with using binary releases directly from Github.

**Pull request merged**